### PR TITLE
Address flakiness of testSSLHandshakeErrorPropagation

### PR DESCRIPTION
**Pull request by Lukasa**
Motivation:

Flaky tests are bad.

This test is flaky because the server closes the connection immediately
upon channelActive. In practice this can mean that the handshake never
even gets a chance to start: by the time the SSLHandler ends up
in the pipeline the connection is already dead. Heck, by the time we
attempt to complete the connection the connection might be dead.

Modifications:

- Change the shutdown to be on first read.
- Remove the disabled autoRead.
- Change the expected NIOTS failure mode to connectTimeout,
    which is how this manifests in NIOTS.

Result:

Test is no longer flaky.

### CODE REVIEW

**Review by artemredkin — APPROVED** [review]

**Review by weissi — APPROVED** [review]
Thank you! LGTM

**Comment by Lukasa** [comment]
Oh hey, fun, a _new_ flaky test! Tracking in #290.

**Comment by Lukasa** [comment]
@swift-server-bot test this please

**Comment by Lukasa** [comment]
@weissi please re-review, I added a new test.

File: Tests/AsyncHTTPClientTests/HTTPClientTests.swift [hunk 1]
```diff
@@ -2697,15 +2696,71 @@ class HTTPClientTests: XCTestCase {
...
+        let task = client.execute(request: request, delegate: TestHTTPDelegate())
+
+        XCTAssertThrowsError(try task.wait()) { error in
+            #if os(Linux)
+                XCTAssertEqual(error as? NIOSSLError, NIOSSLError.uncleanShutdown)
+            #else
+                if isTestingNIOTS() {
+                    XCTAssertEqual(error as? ChannelError, .connectTimeout(.milliseconds(100)))
+                } else {
+                    XCTAssertEqual((error as? IOError).map { $0.errnoCode }, 54)
```
Comment (weissi) [inline Tests/AsyncHTTPClientTests/HTTPClientTests.swift:2717]: any reason to use `54` over `ECONNRESET`?

**Review by weissi — APPROVED** [review]
thanks @Lukasa , looks mostly good to me, just one question around hardcoding 54 for ECONNRESET

File: Tests/AsyncHTTPClientTests/HTTPClientTests.swift (hunk 1 above)
Comment (Lukasa) [reply Tests/AsyncHTTPClientTests/HTTPClientTests.swift:2717]: I just cribbed it from the test above, happy to swap it.

**Review by weissi — APPROVED** [review]
awesome, thank you!

**Comment by Lukasa** [comment]
@swift-server-bot test this please

**Comment by Lukasa** [comment]
One good validation of this having fixed a flaky test is that while we've hit at least two _other_ flaky tests, we've not hit the one this was trying to fix!

**Pull request merged**